### ISSUE TITLE
Implement `llvm_points_to_untyped` in SAW Python client

### ISSUE THREAD

**Issue by RyanGlScott** [GaloisInc/galois-py-toolkit#10]
This would be useful for the sorts of programs characterized in GaloisInc/saw-script#1081.

**Comment by jldodds** [comment]
I believe a reasonable test case would be changing any existing points-to to an untyped points-to. Bonus points if the types actually don't match, but that shouldn't be strictly required

**Comment by pnwamk** [comment]
Is it true that any points-to can be changed to an untyped-points-to but not vice versa? (Where the bonus points case falls under the the can be untyped but not typed?)

**Comment by jldodds** [comment]
Yeah, exactly. Untyped can be used for something with appropriate size but wrong type. An example would be saying byte sequence of appropriate size points to a struct. The text of the issue @RyanGlScott linked actually contains a full example (although it doesn't currently work for other reasons)

**Comment by RyanGlScott** [comment]
Some further discussion on GaloisInc/saw-script#1081 reveals that `llvm_points_to_at_type` is more apt for our use case. That being said, my understanding is that both `llvm_points_to_at_type` and `llvm_points_to_untyped` are thin wrappers around the [same function](https://github.com/GaloisInc/saw-script/blob/dbcccd31062fb0bfeba61af8348df144bb94263b/src/SAWScript/Crucible/LLVM/Builtins.hs#L2021-L2062) in SAW, so perhaps we can extend `llvm.points_to` in a way that makes it possible to emulate both `_at_type` and `_untyped`.

Another variant of `llvm_points_to` that we might consider is `llvm_conditional_points_to`, as well as its siblings `llvm_conditional_points_to_at_type` and `llvm_conditional_points_to_untyped`.

**Comment by RyanGlScott** [comment]
I think the [SAW remote API](https://github.com/GaloisInc/saw-script/blob/master/saw-remote-api/docs/SAW.rst#specifications) for `{pre,post} points to` itself is missing the necessary functionality we'd need for these new variants of `llvm_points_to`. Currently, the remote API only allows specifying the pointer and the pointee, whereas [the function that implements `llvm_points_to` internally in SAW](https://github.com/GaloisInc/saw-script/blob/dbcccd31062fb0bfeba61af8348df144bb94263b/src/SAWScript/Crucible/LLVM/Builtins.hs#L2021-L2062) has three extra arguments:

```hs
llvm_points_to_internal ::
  Bool {- ^ whether to check type compatibility -} ->
  Maybe L.Type {- ^ optional type constraint for rhs -} ->
  Maybe TypedTerm ->
  AllLLVM SetupValue {- ^ lhs pointer -} ->
  AllLLVM SetupValue {- ^ rhs value -} ->
  LLVMCrucibleSetupM ()
```

Perhaps we should add these three extra arguments to the SAW remote API as well.

**Comment by RyanGlScott** [comment]
I've implemented a prototype for how this might look in [this PR](https://github.com/GaloisInc/galois-py-toolkit/pull/12), which in turn requires [these extensions to the SAW remote API](https://github.com/GaloisInc/saw-script/pull/1082). I haven't been able to thoroughly test this yet, as I couldn't get the `tests/saw` tests under `galois-py-toolkit` to run, but my understanding is that that is something that @pnwamk is working on.

**Comment by RyanGlScott** [comment]
Implemented in #12.
